app/src/main/java/com/example/finalproject2/firebase/database/FirebaseDatabaseImp.kt

```diff
@@ -0,0 +1,109 @@
+package com.example.finalproject2.firebase.database
+
+import com.example.finalproject2.model.Friend
+import com.example.finalproject2.model.FriendResponse
+import com.example.finalproject2.model.User
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+import io.reactivex.Observable
+import javax.inject.Inject
+
+
+class FirebaseDatabaseImp @Inject constructor(private var firebaseDatabase: FirebaseDatabase) :
+    FirebaseDatabaseInterface {
+
+    override fun createUser(
+        id: String,
+        name: String,
+        email: String
+    ): Observable<Void> {
+        return Observable.create { emitter ->
+            val user = User(id, name, email)
+            firebaseDatabase.reference.child(CHILD_USER_PATH).child(id).setValue(user)
```

Comment:
dùng setValue(user) thì firebase nó sẽ lưu luôn các fields có trong model ```User```, nếu chỉ lưu một vài trường nhất định thì dùng ```updateChildren()``` nha
https://firebase.google.com/docs/database/android/read-and-write#update_specific_fields